orchagent/portsorch.cpp

```diff
@@ -928,9 +959,9 @@ bool PortsOrch::initPort(const string &alias, const set<int> &lane_set)
 
                 std::string delimiter = "";
                 std::ostringstream counters_stream;
-                for (int cntr = SAI_PORT_STAT_IF_IN_OCTETS; cntr <= SAI_PORT_STAT_PFC_7_ON2OFF_RX_PKTS; ++cntr)
+                for (auto it = portStatIds.begin(); it !=  portStatIds.end(); it++)
```

Comment:
for (const auto& id: portStatIds) // Shorter version
{
}
if you have a container with a primitive type we could write it shorter
for(auto id: portStatIds)
{
}

Another suggestion is never use postfix increment/decrement. In these cases on every operation the object will be copied by its copy constructor.

P.S.: My both comments doesn't mean you must change your code. They're only suggestions.

Reply:
yes, it is much easier to read.